src/analytics/address_balance.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use crate::types::{stardust::block::{Address, Output}, tangle::MilestoneIndex};
+
+use super::TransactionAnalytics;
+
+pub struct AddressCount(usize);
+
+struct AddressBalanceAnalytics {
+    addresses: HashSet<Address>,
+}
+
+impl TransactionAnalytics for AddressBalanceAnalytics {
+    type Measurement = AddressCount;
+
+    fn begin_milestone(&mut self, _: MilestoneIndex) {}
+
+    fn handle_transaction(&mut self, inputs: &[Output], outputs: &[Output]) {
+        for input in inputs {
+            if let Some(a) =  input.owning_address() {
+                self.addresses.remove(a);
+            }
+        }
+
+        for output in outputs {
+            if let Some(a) =  output.owning_address() {
+                self.addresses.insert(a.clone());
+            }
+        }
+    }
+
+    fn end_milestone(&mut self, index: MilestoneIndex) -> Option<Self::Measurement> {
```

Comment:
Why the `Option`? Isn't it safe to assume that whenever `end_milestone` is called the measurement has been created?

Reply:
Oh, I think I got it ... the idea is to skip intermediate incremental states in which we're not interested in.